tests/de.ovgu.featureide.fm.ui-test/src/de/ovgu/featureide/fm/ui/editors/FeatureCreationAndRenamingTest.java

```diff
@@ -0,0 +1,126 @@
+import de.ovgu.featureide.fm.ui.editors.featuremodel.editparts.FeatureEditPart;
+
+/**
+ * This class tests that in a feature model, new features can be created above, below existing features or as siblings, and that they can be renamed right after
+ * their creation.
+ *
+ * @author Jedelhauser Kevin
+ * @author Jutz Benedikt
+ */
+public class FeatureCreationAndRenamingTest {
+
+	/**
+	 * The feature model to edit.
+	 */
+	private FeatureModelManager testModelManager;
+	/**
+	 * The feature diagram editor.
+	 */
+	private FeatureDiagramEditor editor;
+	private IGraphicalFeatureModel gfm;
+
+	/**
+	 * Load the feature model under "./testFeatureModels/basic.xml", and create a new {@link FeatureDiagramEditor} to display it in.
+	 */
+	@Before
+	public void loadBasicModel() {
+		testModelManager = Commons.loadTestFeatureModelFromFile("basic.xml");
+		gfm = new GraphicalFeatureModel(testModelManager);
+		gfm.init();
+		editor = new FeatureDiagramEditor(testModelManager, gfm, true);
+		editor.setFocus();
+		editor.getViewer().setContents(gfm);
+	}
+
+	/**
+	 * Test the creation of new features: <ol> <li>Select the feature named "A", and create a feature below named "C".</li> <li>Select the new feature "C", and
+	 * create a sibling feature named "D".</li> <li>Select feature "B", and create a feature above named "E".</li> </ol>
+	 */
+	@Test
+	public void testCreationMethods() {
+		// Get the graphical model.
+		final IGraphicalFeatureModel graphicalModel = editor.getGraphicalFeatureModel();
+
+		final IFeatureModel model = (IFeatureModel) editor.getFeatureModel().getObject();
+
+		// In the editor, select the feature called "A".
+		final IFeature aFeature = model.getFeature("A");
+
+		// Create new feature "C"
+		final IFeature cFeature = new Feature(model, "C");
+		// True if successfully added
+		// model.addFeature(cFeature);
+
+		// select(editor.getViewer().getEditPartRegistry(), model, "A");
+
+		// final SelectionAction selectAction = new SelectionAction(gfm);
+		// selectAction.selectionChanged(new SelectionChangedEvent());
+
+		// TODO NullPointerException wahrscheinlich, weil das Feature "C" nicht ausgewählt wurde
+		// Siehe dazu: feature in SelectionAction
+		final CreateFeatureBelowAction action = new CreateFeatureBelowAction(editor.getViewer(), gfm);
+		action.run();
+
+		// final FeatureIDEEvent event = new FeatureIDEEvent(model, EventType.FEATURE_ADD, aFeature, cFeature);
+		// Create a new FeatureIDEEvent that adds a new feature to the parent feature "A" for the given feature model.
+		// Add the feature "C" below it.
+		// editor.propertyChange(event);
+
+		// Assert that "C" exists and has "A" as parent.
+		final IGraphicalFeature cGraphicalFeature = graphicalModel.getGraphicalFeature(cFeature);
+
+		// Does feature "C" exist?
+		assertNotNull(cGraphicalFeature);
+
+		// Is feature "C" child of feature "A"?
+		assertTrue(graphicalModel.getGraphicalFeature(aFeature).getGraphicalChildren().contains(cGraphicalFeature));
+	}
+
+	/**
```

Comment:
Params in Javadoc are not correct anymore.
In general, either provide more information or remove Javadoc if it is not required.